lib/oneview-sdk/resource/api300/c7000/server_hardware.rb

```diff
@@ -75,6 +76,35 @@ def patch(operation, path, value = nil)
           @client.response_handler(response)
         end
 
+        # Add one scope to the enclosure
+        # @param [OneviewSDK::API300::C7000::Scope] scope The scope resource
+        # @raise [OneviewSDK::IncompleteResource] if the uri of scope is not set
+        def add_scope(scope)
+          scope.ensure_uri
+          patch('add', '/scopeUris/-', scope['uri'])
+        end
+
+        # Remove one scope from the enclosure
+        # @param [OneviewSDK::API300::C7000::Scope] scope The scope resource
+        # @raise [OneviewSDK::IncompleteResource] if the uri of scope is not set
+        def remove_scope(scope)
+          scope.ensure_uri
+          scope_index = @data['scopeUris'].find_index { |uri| uri == scope['uri'] }
+          patch('remove', "/scopeUris/#{scope_index}", nil) if scope_index
+        end
+
+        # Change the list of scopes in the enclosure
+        # @param [Array[OneviewSDK::API300::C7000::Scope]] scopes The scopes list (or scopes separeted by comma)
+        # @raise [OneviewSDK::IncompleteResource] if the uri of each scope is not set
+        def replace_scopes(*scopes)
+          scopes = scopes.flatten
+          uris = scopes.map do |scope|
```

Comment:
This scopes flatten and map is repeated all over the code. Please create a helper method to it.